### PR TITLE
Batch session recording events

### PR DESCRIPTION
**Pull request by macobo**
## Changes

Session recording events are sent separately from rest of batches to make sure transport options are enforced and we now send events every 300ms when session recording is on (to avoid dropped events).

## Checklist
- [x] Tests for new code (if applicable)
- [x] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/request-queue.js [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-const POLL_INTERVAL = 3000
-
 export class RequestQueue {
-    constructor(handlePollRequest) {
+    constructor(handlePollRequest, pollInterval = 3000) {
         this.handlePollRequest = handlePollRequest
         this.isPolling = true // flag to continue to recursively poll or not
         this._event_queue = []
         this._empty_queue_count = 0 // to track empty polls
+        this.pollInterval = pollInterval
```
Comment (mariusandra) [inline src/request-queue.js:9]: Why both `this._pollInterval` and `this.pollInterval`?

File: src/request-queue.js [hunk 2]
```diff
@@ -53,24 +61,26 @@ export class RequestQueue {
...
     }
 
     formatQueue() {
         const requests = {}
         _.each(this._event_queue, (request) => {
-            const { url, data } = request
-            if (requests[url] === undefined) requests[url] = []
-            requests[url].push(data)
+            const { url, data, options } = request
+            const key = options?._batchKey || url
```
Comment (mariusandra) [inline src/request-queue.js:81]: This line gave me some problems:

image

What's the chance `options` is not an object? When I tried, 3 tests failed and there seemed to be a rabbit hole if I defaulted it to `{}` in `enqueue`.

There's another rabbit hole to go down in if you'd like to enable babel to transpile this into slightly older JS... though I'd just bail and `? :` here..

**Review by mariusandra — CHANGES_REQUESTED** [review]
Found two comments. After quickly pathing them locally while testing everything else worked!

File: src/request-queue.js (hunk 2 above)
Comment (macobo) [reply src/request-queue.js:81]: Interesting.

`options` definetly may not be an object, in this case jest tests actually tested this and didn't catch an error.

Updated it, thanks!

**Review by mariusandra — APPROVED** [review]
LGTM!

**Pull request merged**